Cypress/ApiTesting.md

```diff
@@ -0,0 +1,102 @@
+- Manual Tests
+- Nightly builds
+
+### Prerequisites And Framework
+
+We use the End to End Testing Framework - [**Cypress**](https://docs.cypress.io/guides/overview/why-cypress) for testing Litmus API.
+As Cypress is a npm package, for writing tests developers have to install LTS version of npm before writing tests.
+
+### Setup
+
+- Firstly we need a k8s cluster to deploy the litmus portal, users can use any K8s Cluster. After that, you need to run the following command to install portal:
+
+```
+kubectl apply -f https://raw.githubusercontent.com/litmuschaos/litmus/master/litmus-portal/manifests/cluster-k8s-manifest.yml
+```
+
+- Clone the [Litmus-e2e]("https://github.com/litmuschaos/litmus-e2e") repository, checkout to your branch and change directory to Cypress -
+
+```
+git clone https://github.com/litmuschaos/litmus-e2e.git && cd litmus-e2e
+git checkout -b <YOUR_BRANCH_NAME>
+cd Cypress
+```
+
+- As we use different plugins from Cypress for testing different Scenarios, we have to install all packages. For this Developers can use npm -
+
+```
+npm install
+```
+
+- Once all packages are installed, check if all the pods are running.
+
+```
+watch kubectl get pods -n litmus
+```
+
+- After all the pods come in **Running** state, port forward the pods by running each of the below commands in a different terminal. 
+```
+kubectl port-forward svc/litmusportal-auth-server-service 3000:9003 -n litmus 
+kubectl port-forward svc/litmusportal-server-service 9002:9002 -n litmus
+kubectl port-forward svc/litmusportal-frontend-service 3001:9091 -n litmus
+kubectl port-forward svc/mongo-service 27017:27017 -n litmus
+```
+
+- Now, we can start Cypress Test Runner using command -
+
+```
+npx cypress open --config-file cypress.prod.json
+```
+
+Optional:- For running tests related to GitOps, [create a GitHub Personal Access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) and add that in `githubToken` field in `cypress.prod.json`.
+
+- Developers can manually trigger the tests from the Cypress Test Runner by clicking on test names and play around to get started.
+
+Whenever we add a new test, it will be visible in Cypress Test Runner.
+
+### Add a test in Portal API Pipeline
+
+If you're writing a _new test_ , you will have to add it inside the `cypress/integrations/Api_Tests` directory.
+
+**Currently, we have 2 categories of API -**
+
+- **[REST API](https://litmuschaos.github.io/litmus/auth/v2.0.0/api.html)** for Litmus Portal Authentication
+
+- **[GraphQL API](https://litmuschaos.github.io/litmus/graphql/v2.9.0/api.html)** for ChaosCenter
+
+If you are adding a new category, than create a directory according to test case and add the same as a job in github actions [workflow](https://github.com/litmuschaos/litmus-e2e/actions/workflows/portal-pipeline.yml)
```

Comment:
The link should be - https://github.com/litmuschaos/litmus-e2e/blob/master/.github/workflows/portal-api-pipeline.yml